rust/src/services/fetchers/mask_length.rs

```diff
@@ -0,0 +1,35 @@
+use std::task::{Context, Poll};
+
+use futures::future::{self, Ready};
+use tower::Service;
+
+use crate::state_machine::events::{EventListener, EventSubscriber, MaskLengthUpdate};
+
+pub struct MaskLengthRequest;
+pub type MaskLengthResponse = Option<usize>;
+
+/// A service that serves the mask_length for the current round.
+pub struct MaskLengthService(EventListener<MaskLengthUpdate>);
+
+impl MaskLengthService {
+    pub fn new(events: &EventSubscriber) -> Self {
+        Self(events.mask_length_listener())
+    }
+}
+
+impl Service<MaskLengthRequest> for MaskLengthService {
+    type Response = Option<usize>;
```

Comment:
the aliased type may be used here directly. similarly for some of the other fetchers (some of them already use it).

```suggestion
    type Response = MaskLengthResponse;
```